### PR TITLE
packaging/fedora, tests/lib/prepare-restore: helper tool for packing sources for RPM

### PR DESCRIPTION
**Pull request by bboozzoo**
Couple of patches I kept in a branch that adds support for https://packit.dev/ 

The purpose of the helper tool is to create a source archives in a format that Fedora RPM spec (and openSUSE one too) expect. Eventually I would like to use that inside a packit declaration file, which looks like this:

```
specfile_path: packaging/fedora/snapd.spec
synced_files:
  - .packit.yaml
  - packaging/fedora/snapd.spec
upstream_project_name: snapd
downstream_package_name: snapd
create_tarball_command: ["packaging/fedora/pack-source", "-o", "packaging/fedora", "-g"]
current_version_command: ["git", "describe"]
```

The tool is also useful when building binary RPM on a distro where rpmbuild does not have `--build-in-place`, eg. CentOS/RHEL 7 and Amazon Linux 2.

### CODE REVIEW

File: packaging/fedora/pack-source [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+if [ ! -e packaging/fedora/snapd.spec ]; then
+    echo "must be executed at the top of srcdir"
+    exit 1
+fi
+
+outdir=.
+single=0
+
+while getopts "v:o:gsh" arg; do
```
Comment (zyga) [inline packaging/fedora/pack-source]: Such a chance to support `go:sh` ;)

File: packaging/fedora/pack-source [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        h|*)
+            show_help
+            ;;
+    esac
+done
+
+set -x
+
+if [ -z "$version" ]; then
+	# try to read if tform spec
```
Comment (zyga) [inline packaging/fedora/pack-source]:
```suggestion
	# Try to read version from the spec file.
```

File: packaging/fedora/pack-source [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	version="$(awk '/^Version: / { print $2}' packaging/fedora/snapd.spec)"
+fi
+
+tmpdir=$(mktemp -d)
+
```
Comment (zyga) [inline packaging/fedora/pack-source]: Nitpick: `trap`-based cleanup missing?

File: packaging/fedora/pack-source [hunk 4]
```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
```
Comment (zyga) [inline packaging/fedora/pack-source]: Do you need bash for real or just a habit?

File: packaging/fedora/pack-source [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            show_help
+            ;;
+    esac
+done
+
+set -x
+
+if [ -z "$version" ]; then
+	# try to read if tform spec
+	version="$(awk '/^Version: / { print $2}' packaging/fedora/snapd.spec)"
```
Comment (zyga) [inline packaging/fedora/pack-source]: Hmm, this is hard-coding fedora spec in the world where we have an openSUSE spec as well. This feels wrong especially that we also use this script on openSUSE.

Comment (bboozzoo) [reply packaging/fedora/pack-source]: As discussed on IRC, this can be dropped now.

File: packaging/fedora/pack-source (hunk 3 above)
Comment (bboozzoo) [reply packaging/fedora/pack-source]: Added

File: packaging/fedora/pack-source (hunk 4 above)
Comment (bboozzoo) [reply packaging/fedora/pack-source]: Needed for `trap .. EXIT` now.

**Pull request merged**